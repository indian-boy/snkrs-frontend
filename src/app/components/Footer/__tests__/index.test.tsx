import { render } from '@testing-library/react';
import React from 'react';
import { Footer } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<Footer  />', () => {
  it('should match snapshot', () => {
    const footer = render(<Footer />);
    expect(footer.container.firstChild).toMatchSnapshot();
  });
});
