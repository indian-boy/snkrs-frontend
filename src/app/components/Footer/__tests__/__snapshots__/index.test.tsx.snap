// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer  /> should match snapshot 1`] = `
.c4 {
  width: 100%;
  background: #000000;
  overflow: hidden;
  position: relative;
  color: #FFFFFF;
}

.c4[open] > summary:before {
  background: #FFFFFF;
}

.c4[open] > summary:after {
  background: #D6D6D6;
}

.c4[open] > summary span:after {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c5 {
  display: block;
  background: #000000;
  padding: 1rem 3rem 1rem 0;
  position: relative;
  cursor: pointer;
  font-size: 0.875rem;
  line-height: 1.5rem;
  text-transform: uppercase;
}

.c5:before {
  content: '';
  background: #D6D6D6;
  position: absolute;
  top: 0;
  height: 1px;
  width: calc(100% - 2.5rem);
  left: 0;
}

.c5:after {
  content: '';
  background: #FFFFFF;
  position: absolute;
  bottom: 0;
  height: 1px;
  width: calc(100% - 2.5rem);
  left: 0;
}

.c5::-webkit-details-marker {
  display: none;
}

.c6 {
  cursor: pointer;
}

.c6:before,
.c6:after {
  content: '';
  position: absolute;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
}

.c6:before {
  top: calc(1px - -50%);
  left: calc(100% - 2rem);
  right: 0.7rem;
  height: 0.1rem;
  background: #D6D6D6;
}

.c6:after {
  top: 1.2rem;
  right: 1.3rem;
  bottom: 1rem;
  width: 0.1rem;
  background: #FFFFFF;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  margin: 1rem 1.5rem;
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #F7F7F7;
  font-size: 0.75rem;
  line-height: 1.5rem;
}

.c8:visited {
  color: #D6D6D6;
}

.c0 {
  padding: 2rem 1rem;
  background: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 2rem;
  border-bottom: 1px solid #4D4D4D;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
  text-transform: uppercase;
  font-size: 1rem;
  font-weight: bold;
}

.c2:visited {
  color: #D6D6D6;
}

.c3 {
  padding-bottom: 1rem;
}

.c9 {
  color: #FFFFFF;
  font-weight: bold;
  font-size: 1rem;
  line-height: 1.5rem;
  text-transform: uppercase;
  margin-bottom: 0.875rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
}

.c11 {
  display: grid;
  grid-template-columns: repeat(4,min-content);
  gap: 0.5rem;
  border-bottom: 1px solid #4D4D4D;
  padding-bottom: 2rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  gap: 0.5rem;
}

.c13 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
  font-size: 0.75rem;
  line-height: 1.5rem;
}

.c13:visited {
  color: #D6D6D6;
}

.c14 {
  font-size: 0.75rem;
  line-height: 1.5rem;
  text-align: center;
  color: #D6D6D6;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/lorem-a"
    >
      components.footer.findShoppingStore
      Encontre Uma Loja Nike
    </a>
    <a
      class="c2"
      href="/lorem-b"
    >
      components.footer.registerForNews
      Cadastre-se para receber novidades
    </a>
    <a
      class="c2"
      href="/lorem-c"
    >
      components.footer.siteMap
      Mapa do site
    </a>
  </div>
  <div
    class="c3"
  >
    <details
      class="c4"
    >
      <summary
        class="c5"
      >
        components.footer.help.summary
        Ajuda
        <span
          class="c6"
        />
      </summary>
      <div
        class="c7"
      >
        <a
          class="c8"
          href="/some-url"
        >
          Some link
        </a>
      </div>
    </details>
    <details
      class="c4"
    >
      <summary
        class="c5"
      >
        components.footer.aboutNike.summary
        Sobre a Nike
        <span
          class="c6"
        />
      </summary>
      <div
        class="c7"
      >
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.generalQuestions
          Dúvidas Gerais
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.findYourSize
          Encontre seu Tamanho
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.orders
          Pedidos
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.exchangesAndReturns
          Trocas e Devoluções
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.editConsent
          Editar consentimento
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.payments
          Pagamentos
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.cookiePreferences
          Preferências de Cookie
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.products
          Produtos
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.corporate
          Corporativo
        </a>
        <a
          class="c8"
          href="/some-url"
        >
          components.footer.aboutNike.contactUs
          Fale Conosco
        </a>
      </div>
    </details>
  </div>
  <div
    class="styles__SocialMedias-sc-1c785za-5"
  >
    <div
      class="c9"
    >
      components.footer.socialMedias
      Redes Sociais
    </div>
    <div
      class="c10"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <svg>
          facebook_rounded_white.svg
        </svg>
      </a>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <svg>
          instagram_rounded_white.svg
        </svg>
      </a>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <svg>
          youtube_rounded_white.svg
        </svg>
      </a>
    </div>
  </div>
  <div
    class="styles__PaymentMethods-sc-1c785za-7"
  >
    <div
      class="c9"
    >
      components.footer.paymentMethods
      Formas de pagamento
    </div>
    <div
      class="c11"
    >
      <svg>
        amex.svg
      </svg>
      <svg>
        discover.svg
      </svg>
      <svg>
        elo.svg
      </svg>
      <svg>
        hipercard.svg
      </svg>
      <svg>
        mastercard.svg
      </svg>
      <svg>
        visa.svg
      </svg>
    </div>
  </div>
  <div
    class="c12"
  >
    <a
      aria-current="page"
      class="c13 active"
      href="/"
    >
      Brasil
    </a>
    <a
      aria-current="page"
      class="c13 active"
      href="/"
    >
      components.footer.paymentMethods
      Política de Privacidade
    </a>
    <a
      aria-current="page"
      class="c13 active"
      href="/"
    >
      components.footer.userTerms
      Termos de Uso
    </a>
  </div>
  <span
    class="c14"
  >
    components.footer.companyInfos
    © 2021 Nike. Todos os direitos reservados. Fisia Comércio de Produtos Esportivos Ltda - CNPJ: 59.546.515/0016-10 Rua José Semião Rodrigues Agostinho, 1370 - Galpão Unidade 17 - Nível 0, 1 e 2 - CEP 06833-300 - Embu das Artes/SP.
  </span>
</div>
`;
